scripts/product_revision_to_historical_events.pl

```diff
@@ -119,17 +134,44 @@ sub find_products {
 		next if $entry =~ /^\.\.?$/;
 		my $file_path = "$dir/$entry";
 		if (-d $file_path) {
-			find_products($file_path,"$code$entry");
+			find_products($file_path, "$code$entry");
 			next;
 		}
+
 		if ($entry eq 'product.sto') {
-			process_file($dir);
-		} 
+			if ($can_process or ($last_processed_path and $last_processed_path eq $dir)) {
```

Comment:
readdir() can return files in any order, so you need to add a sort in the foreach my $entry (@files) loop

this would also allow to restart without exploring all the paths again, you could just check if the path matches the start of the path of the last processed product. That way if the last one is 808/400/300/2001, you don't need to go inside all the directories from 001 to 807